src/color_laws.jl

```diff
@@ -228,3 +248,131 @@ function gcc09_invum(x::Real, Rv::Real)
 
     return a + b / Rv
 end
+
+# Shape models used by F99 and F04
+function _curve_F99_method(
+    x,
+    Rv,
+    c1,
+    c2,
+    c3,
+    c4,
+    x0,
+    gamma,
+    f99_optnir_axav_x,
+    optnir_axav_y,
+    )
+
+    # add in required spline points, otherwise just spline points
+    if x >= f99_x_cutval_uv
+        xuv = (f99_x_splineval_uv..., x)
+    else
+        xuv = f99_x_splineval_uv
+    end
+
+    # FM90 model and values
+    fm90_model = FM90(c1=c1, c2=c2, c3=c3, c4=c4, x0=x0, gamma=gamma)
+    # evaluate model and get results in A(x)/A(V)
+    axav_fm90 = @. fm90_model(aa_to_invum(xuv)) / Rv + 1
+
+    # ignore the spline points
+    if x >= f99_x_cutval_uv
+        axav = last(axav_fm90)
+    else
+        # **Optical Portion**
+        # using cubic spline anchored in UV, optical, and IR
+
+        # optical/NIR points in input x
+
+        # save spline points
+        y_splineval_uv = axav_fm90
+
+        # spline points
+        x_splineval_optir = (0.0, f99_optnir_axav_x...)
+
+        # determine optical/IR values at spline points
+        y_splineval_optir = (0.0, optnir_axav_y...)
+        spline_x = (x_splineval_optir..., f99_x_splineval_uv...)
+        spline_y = (y_splineval_optir..., y_splineval_uv...)
+        spl = Spline1D(collect(spline_x), collect(spline_y), k=3)
+        axav = spl(x)
+    end
+
+    # return A(x)/A(V)
+    return axav
+end
+
+"""
+    F99(;Rv=3.1)
+
+Fitzpatrick (1999) dust law.
+
+Returns E(B-V) in magnitudes at the given wavelength relative to the
+extinction. This model applies to the UV and optical to NIR spectral range.
+The default support is [1000, 33333] Å. Outside of that range this will return
+0. Rv is the selective extinction and is valid over [2, 6]. A typical value for
+the Milky Way is 3.1.
+
+# References
+[Fitzpatrick (1999)](https://ui.adsabs.harvard.edu/abs/1999PASP..111...63F/)
+"""
+@with_kw struct F99 <: ExtinctionLaw
+    Rv::Float64 = 3.1
+end
+
+function (law::F99)(wave::T) where T
+    checkbounds(law, wave) || return zero(float(T))
+    x = aa_to_invum(wave)
+    return f99_invum(x, law.Rv)
+end
+
+bounds(::Type{F99}) = (1000.0, 33333.3)
+
+"""
+    DustExtinction.f99_invum(x, Rv)
+
+The algorithm used for the [`F99`](@ref) extinction law, given inverse microns and Rv. For more information, seek the original paper.
+"""
+function f99_invum(x::Real, Rv::Real)
+    if !(0.3 <= x <= 10.0)
+        error("out of bounds of F99, support is over $(bounds(F99)) angstrom")
+    end
+
+    # terms depending on Rv
+    c2 = @evalpoly (1. / Rv) f99_Rv_params...
```

Comment:
This cannot work with this macro, you need to list the elements of the vector one by one. That's why the function was introduced

Reply:
Oh, just saw your comment! Yep, just realized that when I looked at Travis

Reply:
Alrighty, everything should be switched over now. I've never had to think about things like backwards compatibility before, so this has been an extremely useful lesson for me, thanks!